01_classify_stratified/nbconverted/plot_cancer_type_results.py

```diff
@@ -228,3 +228,58 @@ def label_points(x, y, gene, sig, ax):
                            plt.gca())
 adjust_text(text_labels, ax=plt.gca())
 
+
+# ## Confusion matrix
+
+# In[10]:
+
+
+import os
+
+import mpmp.utilities.data_utilities as du
+
+preds_dir = os.path.join(cfg.repo_root, 'results_preds', 'cancer_type')
+sample_info_df = du.load_sample_info()
+
+preds_expression_df = au.load_preds_to_matrix(preds_dir, sample_info_df,
+                                              training_data='expression')
+print(preds_expression_df.shape)
+preds_expression_df.iloc[:5, :5]
+
+
+# In[11]:
+
+
+sns.set({'figure.figsize': (15, 10)})
```

Comment:
* Just to clarify, these confusion matrices are showing how well training on cancer type A (y-axis) can predict cancer type (x-axis)?
* Looks like you have some positive cases that you used to check that there is some gain in information across cancer types. But overall it looks like there isn't much of a gain using different cancer types.
* Looking at the 2 cases where you have cancers that contribute equally to the prediction performance across all cancer types, my intuition says that if these were rare cancers then I would've expected a lower performance. Since the performance is slightly high, this makes me think that there is some common signal in these 2 cancers that is maybe shared across all cancer types...

Reply:
> * Just to clarify, these confusion matrices are showing how well training on cancer type A (y-axis) can predict cancer type (x-axis)?

Exactly. So like I mentioned in my other comment, we're doing one-vs-rest classification of the cancer types. The cancer type on the y-axis is what the model is trained to predict (i.e. the positive label in the training set), and the cancer type on the x-axis is what we're evaluating on (i.e. the true label in the test set). So a high off-diagonal value in the heatmap indicates that a model trained to distinguish the y-axis cancer type from all the others often "confuses" it for the x-axis cancer type (i.e. they have similar gene expression profiles).

> * Looks like you have some positive cases that you used to check that there is some gain in information across cancer types. But overall it looks like there isn't much of a gain using different cancer types.

Right; or stated slightly differently, the models are pretty accurate overall (high on-diagonal values and mostly low off-diagonal values), and the "inaccuracies" (high off-diagonal values) we see aren't surprising based on our knowledge of relationships between cancer types. We expected this result, based on Greg's past work which showed that cancer type prediction is a fairly easy problem for the most part.

> * Looking at the 2 cases where you have cancers that contribute equally to the prediction performance across all cancer types, my intuition says that if these were rare cancers then I would've expected a lower performance. Since the performance is slightly high, this makes me think that there is some common signal in these 2 cancers that is maybe shared across all cancer types...

Yeah, could be! I'm running some experiments now to subsample cancer types (so sample sizes will be uniform across cancers), so if we still see these effects we'll know it's not sample size related.